packages/melody-hooks/src/component.js

```diff
@@ -322,7 +327,14 @@ Object.assign(Component.prototype, {
             // Reset the hooks pointer
             this.hooksPointer = -1;
             // Run the component functions
-            this.data = this.componentFn(this.props) || {};
+
+            try {
+                const dataNext = this.componentFn(this.props) || {};
+                this.data = dataNext;
+            } catch (err) {
+                error = err;
```

Comment:
Instead of rethrowing the error would it be possible to use `finally` to do the cleanup?

Reply:
I thought about that, but wouldn't that involve generator functions somehow when transpiled with babel?

Reply:
but we would have duplicated clean up logic

Reply:
Or you move all of the cleanup logic into the finally block. Its executed regardless of whether there has been an error or not.